shtdlib.sh

```diff
@@ -2227,6 +2223,54 @@ function test_shopt_decorator {
     assert shopt -qo pipefail && color_echo green "Successfully decorated  ${FUNCNAME[0]} with pipefail"
 }
 
+# Test signaling
+function test_signal_process {
+    signal_processor SIGUSR2 'exit 42' > /dev/null
+    local sub_pid_0="${!}"
```

Comment:
I added this because that can fail, this does duplicate some messages but in case of failure this gives an indication of what went wrong with the signal processor